repl/src/main/scala/com/cloudera/livy/repl/sparkr/SparkRInterpreter.scala

```diff
@@ -62,19 +67,37 @@ object SparkRInterpreter {
     ).r.unanchored
 
   def apply(): SparkRInterpreter = {
-    val executable = sparkRExecutable
-      .getOrElse(throw new IllegalStateException("Cannot find sparkR executable."))
+    val backendTimeout = sys.env.getOrElse("SPARKR_BACKEND_TIMEOUT", "120").toInt
+    val mirror = universe.runtimeMirror(getClass.getClassLoader)
+    val sparkRBackendClass = mirror.classLoader.loadClass("org.apache.spark.api.r.RBackend")
+    val ctor = sparkRBackendClass.getDeclaredConstructor().newInstance()
+
+    var sparkRBackendPort = 0
+    val initialized = new Semaphore(0)
+    // Launch a SparkR backend server for the R process to connect to
+    val sparkRBackendThread = new Thread("SparkR backend") {
+      override def run() {
+        sparkRBackendPort = sparkRBackendClass.getMethod("init").invoke(ctor).asInstanceOf[Int]
+        initialized.release()
+        sparkRBackendClass.getMethod("run").invoke(ctor)
+      }
+    }
 
-    val builder = new ProcessBuilder(Seq(executable.getAbsolutePath).asJava)
+    sparkRBackendThread.start()
+    // Wait for RBackend initialization to finish
+    initialized.tryAcquire(backendTimeout, TimeUnit.SECONDS)
 
+    val builder = new ProcessBuilder(Seq("R", "--slave @").asJava)
```

Comment:
now, user can using like this when R is installed custom location
export DRIVER_R=<custom location R>